src/cargo/core/dependency.rs

```diff
@@ -52,52 +52,28 @@ struct Inner {
 }
 
 #[derive(Serialize)]
-struct SerializedDependency<'a> {
-    name: &'a str,
+pub struct SerializedDependency {
+    name: InternedString,
```

Comment:
> I replaced borrowed types in `SerializedDependency` with owned variants to satisfy the borrow checker.
This matches `SerializedPackage` and shouldn't be an issue since `Dependency` is cheap to copy.

Could you help me understand why this didn't borrow check?

One of the goals we have is to eventually migrate all serde types over to `cargo-util-schemas` so people can load the data themselves.  The use of `InternedString` gets in the way of that (granted, we might need to switch to `String` for that...)

Reply:
I had issues with temporaries being dropped in `Dependency::serialized` after copying the code from `Dependency::serialize`. Specificity with [`name`](https://github.com/rust-lang/cargo/blob/dcb4ef9a10630df4fb8b1b2e67072b1267d7d0e3/src/cargo/core/dependency.rs#L84) and [`registry_id`](https://github.com/rust-lang/cargo/blob/dcb4ef9a10630df4fb8b1b2e67072b1267d7d0e3/src/cargo/core/dependency.rs#L82). Looking again, it should be possible to work around this issue.

That being said, I don't think it would be worth changing back. Since the refactor makes `SerializedPackage` contain a `SerializedDependency` rather than a `Dependency` and `SerializedPackage` is an owned type, it would probably be awkward to make `SerializedDependency` require a lifetime, since it would require changes in `SerializedPackage`.

Reply:
Hmm, this all matches `SerializedPackage` so I guess its no worse than where we are at today.